app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -1114,37 +731,72 @@ describe('NetworkController', () => {
                   chainId: '0x1337',
                 },
               },
+              infuraProjectId: 'some-infura-project-id',
             },
-            async ({ controller, network: network1 }) => {
-              network1.mockEssentialRpcCalls();
-              const network2 = new InfuraNetworkCommunications({
-                infuraNetwork: networkType,
-              });
-              network2.mockEssentialRpcCalls();
+            async ({ controller }) => {
+              const fakeProviders = [
+                buildFakeProvider([
+                  {
+                    request: {
+                      method: 'test',
+                    },
+                    response: {
+                      result: 'test response 1',
+                    },
+                  },
+                ]),
+                buildFakeProvider([
+                  {
+                    request: {
+                      method: 'test',
+                    },
+                    response: {
+                      result: 'test response 2',
+                    },
+                  },
+                ]),
+              ];
+              const fakeNetworkClients = [
+                buildFakeClient(fakeProviders[0]),
+                buildFakeClient(fakeProviders[1]),
+              ];
+              mockCreateNetworkClient()
```

Comment:
Here I'm making use of the `jest-when` package to make sure that specific network clients are used for specific invocations of `createNetworkClient`. This is specific to these tests — I don't think any of the tests on the `core` side involve multiple networks in a single test. I aim to copy these tests over to the `core` side to fix that divergence.